teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,129 @@
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
+
+### Use Cases (optional)
+
+- Filter the resources in the Hub UI using platforms.
+- Search the resources with `tkn hub` for specific platform(s).
+- Get information about supported platforms for specific resource in the Hub UI/`tkn hub`.
+
+## Requirements
+
+- Platform information in the catalog resource should be optional.
+- Information how to identify/prove supported platform for the resource should be documented in the Tekton catalog.
+- Tekton Hub should automatically get/update information about platforms of the resources.
+- Hub UI should show all platforms available in database to filter.
+
+## Proposal
+
+### Catalog side
+
+Add new `tekton.dev/platforms` annotation to the resource yaml file with list of platforms separated by comma, for example
+
+```
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/platforms: linux/amd64,linux/s390x
+```
+
+The `tekton.dev/platforms` annotation should be optional. 
+Information about supported platforms also should be added to the `README.md` as separate `Platform` section with extra use case examples for specific platforms if it is required.
+For instance, another value for default image parameter in the task.
+
+The conditions to add platforms into `tekton.dev/platforms` should be written in Tekton catalog documentation.
+The levels of verification are:
+- all the container images, used in the resource, have corresponding platform support.
+- the resource tests were executed on corresponding platform once for conrete version.
+- the resource tests are periodically executed on corresponding platform.
+
+### Hub side
+
+Extend backend logic to be able to pull, parse, store and update information about resource's platforms. The approach is similar to what is available for `tags`.
+If there is no information about platform in the resource definition, then default `linux/amd64` platform should be used.
+
+Hub UI should have `platform` filter and show resources based on it.
+
+CLI code should be extendend:
+
+- to have `--platform` search parameter to filter resources based on platform.
+- to have platform information in the resource information view.
+
+### Tekton CLI side
```

Comment:
This will be on HUB CLI

Reply:
Hmm, what I meant, final user will use tkn cli to get information from the Hub. Should I still call it HUB CLI?

Reply:
tkn depend on hub and bundle the hub cli inside the tkn binary,

which mean you when modify the Hub cli and it will automatically get imported inside "tkn hub"

Reply:
Yep, I know that. I guess text should be reformulated here, to describe that code should be written on hub cli side and will be used in tkn cli.

Reply:
Are the Hub changes to the hub CLI meant to be in scope for this TEP?